dawn/src/dawn/dawn-opt.cpp

```diff
@@ -0,0 +1,188 @@
+    ("t,type", "Type of input [sir,iir]. Deduced if --format=json.", cxxopts::value<std::string>()->default_value("sir"))
+    ("o,out", "Output IIR filename. If unset, writes IIR to stdout.", cxxopts::value<std::string>())
+    ("v,verbose", "Set verbosity level to info. If set, use -o or --out to redirect IIR.")
+    ("p,pass-groups", "Ordered list of pass groups to run. See DawnCompiler.h for list.", cxxopts::value<std::vector<std::string>>())
+    ("h,help", "Display usage.")
+#define OPT(TYPE, NAME, DEFAULT_VALUE, OPTION, OPTION_SHORT, HELP, VALUE_NAME, HAS_VALUE, F_GROUP) \
+  (OPTION, HELP, cxxopts::value<TYPE>()->default_value(toString(DEFAULT_VALUE)))
+#include "dawn/Optimizer/Options.inc"
+#undef OPT
+    ;
+  // clang-format on
+
+  options.parse_positional({"input"});
+
+  const int numArgs = argc;
+  auto result = options.parse(argc, argv);
+
+  if(result.count("help")) {
+    std::cout << options.help() << std::endl;
+    return 0;
+  }
+
+  // Get input = string
+  std::string input;
+  if(result.count("input")) {
+    std::ifstream t(result["input"].as<std::string>());
+    input.insert(input.begin(), (std::istreambuf_iterator<char>(t)),
+                 std::istreambuf_iterator<char>());
+  } else {
+    std::istreambuf_iterator<char> begin(std::cin), end;
+    input.insert(input.begin(), begin, end);
+  }
+
+  // Determine the type (IIR or SIR)
+  const std::string formatString = result["format"].as<std::string>();
+  IRType type = result["type"].as<std::string>() == "sir" ? IRType::SIR : IRType::IIR;
+  if(formatString == "json") {
+    // Deduce representation type
+    auto json = dawn::json::json::parse(input);
+    if(json.count("internalIR") > 0)
+      type = IRType::IIR;
+  }
+
+  dawn::Options dawnOptions;
+#define OPT(TYPE, NAME, DEFAULT_VALUE, OPTION, OPTION_SHORT, HELP, VALUE_NAME, HAS_VALUE, F_GROUP) \
+  dawnOptions.NAME = result[OPTION].as<TYPE>();
+#include "dawn/Optimizer/Options.inc"
+#undef OPT
+  dawn::DawnCompiler compiler(dawnOptions);
+
+  std::map<std::string, std::shared_ptr<dawn::iir::StencilInstantiation>> optimizedSIM;
+
+  std::list<dawn::PassGroup> passGroups;
+  if(result.count("pass-groups") == 0) {
+    passGroups = dawn::DawnCompiler::defaultPassGroups();
+  } else {
+    for(auto pg : result["pass-groups"].as<std::vector<std::string>>()) {
+      passGroups.push_back(parsePassGroup(pg));
+    }
+  }
+
+  // Deserialize
+  if(type == IRType::SIR) {
+    const dawn::SIRSerializer::Format format = formatString == "json"
+                                                   ? dawn::SIRSerializer::Format::Json
+                                                   : dawn::SIRSerializer::Format::Byte;
+    std::shared_ptr<dawn::SIR> stencilIR =
+        dawn::SIRSerializer::deserializeFromString(input, format);
+    auto parallelPos = std::find(passGroups.begin(), passGroups.end(), dawn::PassGroup::Parallel);
+    if(parallelPos != passGroups.end())
+      passGroups.erase(parallelPos);
```

Comment:
The current behavior now is that we don't allow that group to be passed. It's still in the enum type in DawnCompiler.h for now (because CompilerUtil uses that) but once we finish the refactor, CompilerUtil will disappear and Parallel can be removed from the enum too.

Reply:
Great